### PR TITLE
feat: MLIBZ-2361 Implement live service router to integrate with PubNub library

### PR DESCRIPTION
**Pull request by yuliya-guseva**
#### Description
Implement live service router to integrate with pub nub library

#### Changes
Added singleton RealtimeRouter class with initialize and unInitialize methods. 

#### Tests
Will be tested in the MLIBZ-2362 branch with register/unregister methods

### CODE REVIEW

File: java-api-core/src/com/kinvey/java/store/RealtimeRouter.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            if (realtimeRouter == null) {
+                realtimeRouter = new RealtimeRouter();
+            }
+            return realtimeRouter;
+        }
+    }
+
+    void initialize(String channelGroup, String publishKey, String subscribeKey, String authKey, AbstractClient client) {
+        synchronized (lock) {
+            if (!isInitialized()) {
```
Comment (vinaygahlawat) [inline java-api-core/src/com/kinvey/java/store/RealtimeRouter.java]: Do we need to acquire this lock every time initialized is called, even if the `RealtimeRouter` has already been initialized? We may be able to make this more efficient by doing [double-check locking](https://en.wikipedia.org/wiki/Double-checked_locking).

File: java-api-core/src/com/kinvey/java/store/RealtimeRouter.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                pubnubClient.addListener(subscribeCallback);
+                pubnubClient.subscribe().channelGroups(Collections.singletonList(channelGroup)).execute();
+            }
+        }
+
+    }
+
+    void unInitialize() {
+        synchronized (lock) {
+            if (isInitialized()) {
```
Comment (vinaygahlawat) [inline java-api-core/src/com/kinvey/java/store/RealtimeRouter.java]: Same question here about locking as in the `initialize()` method.

File: java-api-core/src/com/kinvey/java/store/RealtimeRouter.java (hunk 1 above)
Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/store/RealtimeRouter.java]: I agree. Updated.

File: java-api-core/src/com/kinvey/java/store/RealtimeRouter.java (hunk 2 above)
Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/store/RealtimeRouter.java]: Updated.

**Comment by yuliya-guseva** [comment]
Changed `RealtimeRouter` to `LiveServiceRouter`.

**Pull request merged**